src/hotspot/share/opto/compile.cpp

```diff
@@ -4453,6 +4453,44 @@ void Compile::print_inlining_commit() {
   print_inlining_reset();
 }
 
+void Compile::print_inlining_append_late(CallGenerator* cg, InliningResult result, const char* msg) {
+  assert(print_inlining() || print_intrinsics(), "PrintInlining off?");
+  C->print_inlining_assert_ready();
+  C->print_inlining_move_to(cg);
+
+  stringStream* old = _print_inlining_list->at(_print_inlining_idx)->ss();
+
+  auto old_size = old->size();
+  bool found_line_break = false;
+  while (!found_line_break && old_size > 0) {
+    if (old->base()[old_size - 1] == '\n') {
+      found_line_break = true;
+    }
+    old_size--;
+  }
+
+  if (old_size == 0) {
+    CallNode* call = cg->call_node();
+    print_inlining(cg->method(), call->jvms()->depth() - 1, call->jvms()->bci(), result, msg);
+    print_inlining_update_delayed(cg);
+  } else {
+    // This is very similar to the previous branch with print_inlining_update_delayed with the difference that
+    // we are copying the old contents without the line break.
+
+    auto buffer = new PrintInliningBuffer();
```

Comment:
This function is a lot more complicated than I expected.  I expected it to look more like Compile::print_inlining_commit() but writing into the existing PrintInliningBuffer().  If we didn't terminate lines in the _print_inlining_list with '\n' then we wouldn't need to remove it before appending.